libs/mep/mwpm-143159/timeline/timeline.js

```diff
@@ -0,0 +1,153 @@
+    sides[index].append(...createBars(index));
+    barRow.append(sides[index]);
+  });
+  return barRow;
+}
+function addBottomRow(periodText) {
+  const [periodRow, left, right] = createRow();
+  const sides = [left, right];
+  periodText.forEach((text, index) => {
+    sides[index].append(createTag('p', { class: 'period body-s' }, text));
+    periodRow.append(sides[index]);
+  });
+  return periodRow;
+}
+
+function setBG(el, color) {
+  el.style.background = color;
+}
+
+function updateForRTL(lgStr, el) {
+  if (el.closest('[dir="rtl"]')) {
+    return lgStr.replace(/to right|to left|90deg|180deg|turn 0.25|turn 0.75/g, (match) => {
+      const converter = {
+        'turn 0.25': 'turn 0.75',
+        'turn 0.75': 'turn 0.25',
+        '90deg': '180deg',
+        '180deg': '90deg',
+        'to right': 'to left',
+        'to left': 'to right',
+      };
+
+      return converter[match];
+    });
+  }
+  return lgStr;
+}
+function setColors(colors, fragment, el) {
+  const barEls = fragment.querySelectorAll('.bar');
+  const periodEls = fragment.querySelectorAll('.period');
+  if (colors?.length === 2 && isColorOrGradient(colors[0]) && isColorOrGradient(colors[1])) {
+    if (barEls.length === 3 || periodEls.length === 2) {
+      const leftColor = colors[0];
+      const rightColor = colors[1];
+      let firstBar; let secondBar; let thirdBar;
+      if (isGradient(leftColor)) {
+        leftColor.split(' ').forEach((color) => {
+          if (isColor(color) && !firstBar) {
+            firstBar = color;
+            setBG(barEls[0], firstBar);
+          } else if (isColor(color)) {
+            secondBar = color;
+            setBG(barEls[1], secondBar);
+          }
+        });
+      } else {
+        setBG(barEls[0], leftColor);
+        setBG(barEls[1], leftColor);
+      }
+      if (isGradient(rightColor)) {
+        leftColor.split(' ').forEach((color) => {
+          if (isColor(color) && !thirdBar) {
+            thirdBar = color;
+            setBG(barEls[2], thirdBar);
+          }
+        });
+      } else {
+        setBG(barEls[2], rightColor);
+      }
+      setBG(periodEls[0], updateForRTL(leftColor, el));
+      setBG(periodEls[1], updateForRTL(rightColor, el));
+    }
+  }
+}
+
+function colWidthsNotValid(colWidths) {
+  return (colWidths.length !== 2 || colWidths.some((value) => Number.isNaN(value)));
+}
+function updateColWidths(colWidths, fragment) {
+  if (colWidthsNotValid(colWidths)) return;
+  const total = Number(colWidths[0]) + Number(colWidths[1]);
+  const right = Math.floor((Number(colWidths[1]) / total) * 10000) / 100;
+  const colString = `1fr minmax(${String(right)}%, 150px)`;
+  fragment.querySelectorAll('.row').forEach((row) => {
+    row.style.gridTemplateColumns = colString;
+  });
+}
+export default function init(el) {
+  const fragment = document.createDocumentFragment();
+  const [textRow, left, right] = createRow();
+  const rows = el.querySelectorAll(':scope > div > div');
```

Comment:
wouldn't this give you the columns in the rows, not the rows?

Reply:
@vgoodric, this is targeting the content in each of the table rows:
![Screenshot 2024-04-02 at 2 07 08 PM](https://github.com/adobecom/milo/assets/124626043/44745e6b-84cf-4c39-97c2-58c39a7c10a6)